test/extended/etcd/leader_changes.go

```diff
@@ -0,0 +1,31 @@
+package etcd
+
+import (
+	"context"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"github.com/prometheus/common/model"
+
+	"github.com/openshift/origin/test/extended/prometheus/client"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("etcd", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLI("etcd-leader-change", exutil.KubeConfigPath())
+
+	g.It("leader changes are not excessive", func() {
+		prometheus, err := client.NewE2EPrometheusRouterClient(oc)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		g.By("Watching the number of etcd leadership changes for the next five minutes")
+		// check max(etcd_server_leader_changes_seen_total) every 30s for 5m.
+		// the value should consistently be 1.
+		o.Consistently(func() model.SampleValue {
```

Comment:
This is really a post condition test, or a snapshot test.  Instead, you should be doing

```
increase((max by (job) (etcd_server_leader_changes_seen_total) or 0*absent(etcd_server_leader_changes_seen_total))[15m:1m]) > 0
```

Reply:
You shouldn't be waiting for 5 minutes.

Reply:
Updated

Reply:
> This is really a post condition test, or a snapshot test. Instead, you should be doing
> 
> ```
> increase((max by (job) (etcd_server_leader_changes_seen_total) or 0*absent(etcd_server_leader_changes_seen_total))[15m:1m]) > 0
> ```

I have nothing against it, but what does this say?  How can `0*` anything be not zero?